### PR TITLE
fix(App): allow `#triggerAction` to receive action and options as array

### PR DESCRIPTION
**Pull request by marstamm**
### Proposed Changes
related to https://github.com/camunda/camunda-modeler/pull/4568

I noticed that with the plugin, the new file dialog from the OS menu does not work. Start page and client side new-tab worked, as we added the special action handling there.

With this change, we allow action and options to be an array for `#triggerAction`, so menu actions can easily add options, even when triggered via a backend module.

image

Try it out:

- Download RPA Plugin here: http://github.com/camunda/camunda-modeler-rpa-plugin/zipball/latest/
- add it as a plugin
- try to open a new `.robot` file from the file menu
- It now works

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [ ] __Visual demo__ attached
* [x] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: client/src/app/App.js [hunk 1]
```diff
@@ -1751,6 +1752,13 @@ export class App extends PureComponent {
       activeTab
     } = this.state;
 
+    // Allows menu actions to easily pass options
+    if (isArray(action)) {
+      options = action[1];
+      action = action[0];
+    }
+
+    console.log(action, options);
```
Comment (nikku) [inline client/src/app/App.js:1761]: Get rid of this
```suggestion
    console.log(action, options);
```

Comment (marstamm) [reply client/src/app/App.js]: 🤦 

Fixed in [88646ed](https://github.com/camunda/camunda-modeler/pull/4575/commits/88646ed5bfe24254c722d240ab578ad8f98a7015)

**Comment by nikku** [comment]
Could you elaborate what exactly is missing from the current way to trigger menus? 

I verified via https://github.com/camunda/camunda-modeler/pull/4575/commits/116337166092fc032b4789eb1b3f7bda6373a45a and https://github.com/camunda/camunda-modeler/pull/4575/commits/5fc71cc1df6940f714b435f71abd581b17173032 to see that we can already pass context.

I propose that we update the existing examples / plug-ins to feature the new extension point / usage patterns within the DEV environment. This makes it simpler for you to test things, but also allows us to verify that we don't break E2E scenarios in the future.

**Comment by marstamm** [comment]
You are right, this is not the correct way to implement this. The problem is that we use the tab provider both the frontend (right plus button) as well as the backend (File menu). 

The Frontend implementation does not support adding options, the backend does. This needs to be fixed in the frontend code that renders the Menu, as well as add the option for "options" in the Slot on the front page. I will revisit this PR

**Comment by marstamm** [comment]
I adjusted the code to accept options for triggered actions and added a simple sample tab plugin that shows a hello world page.

Recording 2024-10-01 at 15 49 45

With this change, you can open new Tabs and trigger events properly from all places we need them

**Comment by nikku** [comment]
> The Frontend implementation does not support adding options, the backend does. This needs to be fixed in the frontend code that renders the Menu, as well as add the option for "options" in the Slot on the front page. I will revisit this PR

What we could consider for a good integration:

* Render `+` and options based off registered tab providers.
* Ensures that UX is always decent (i.e. we could hide ROBOT and friends behind a `...` on the landing page
* New file types are always represented in _all_ relevant areas, just because they are registered.

The current way of doing things is error-prone.

**Comment by marstamm** [comment]
> Render + and options based off registered tab providers.
We already do this

> hide ROBOT and friends behind a ... on the landing page
I would not hide them, we have enough space on the landing page and I do not see us having a lot of additional Providers. If we get to the point, we can reiterate and hide them.

> New file types are always represented in all relevant areas, just because they are registered.
The only place we do not do that yet is the Welcome page. I added this in [7ee2c34](https://github.com/camunda/camunda-modeler/pull/4575/commits/7ee2c34395804fb989359de4a1517b5800686972)

**Comment by nikku** [comment]
https://github.com/camunda/camunda-modeler/pull/4575/commits/7ee2c34395804fb989359de4a1517b5800686972 is great, because it sets us up to do whatever optimizations on the welcome page in the future. Good stuff.

**Review by nikku — APPROVED** [review]
I struggled to get this up and running just to realize that we did not check-in the test-plugin distro (as we do with other distributions). Fixed via https://github.com/camunda/camunda-modeler/pull/4575/commits/d1f280b99221ff2ea60788c99836e72778224220.

Then I checked the welcome page designs. Via  https://github.com/camunda/camunda-modeler/pull/4575/commits/016fe71ea4c51938bb40ba60e4e727dbd17e4d0f I made sure that the UI scales, independent on how many options we offer.

The overall changes look good to me.

**Review by nikku — COMMENTED** [review]
You could do another round of cleanup on the commit history, but otherwise it looks good to merge.

**Comment by marstamm** [comment]
Also noticed that we need `key` attribute so React plays nicely. Will make the adjustments and squash the commits

**Review by nikku — CHANGES_REQUESTED** [review]
It seems like the new way of declaring the `EmptyTab` is incompatible with our `ref`-based approach, cf. errors thrown in the developer console:

screenshot EGuwtO

**Comment by marstamm** [comment]
Fixed via [24e086a](https://github.com/camunda/camunda-modeler/pull/4575/commits/24e086a23246671f01ffa63e5fd31f36941f29fc). As the functions called on the ref are all implemented by the child, we are fine to simply forward the ref to the EmptyTab component.

**Comment by nikku** [comment]
Thanks for the update. The change still did not fully work, as `triggerAction` in your proposed fix was not implemented on the HelloWorld tab. What I propose instead is to implement a proper tab there: https://github.com/camunda/camunda-modeler/pull/4575/commits/45be9f005bc6cfaf71baa600c45dfa06b8577939.

**Comment by nikku** [comment]
Once CI passes I'll squash the changes and merge this PR. :tada:

**Review by nikku — APPROVED** [review]

**Pull request merged**